src/main.go

```diff
@@ -65,8 +65,27 @@ func main() {
 
 	// Register as a Worker service with gRPC and start accepting connections.
 	s := grpc.NewServer()
-	pb.RegisterWorkerServer(s, &foremanServer{jobStorage: newJobStorage()})
+
+	js := newJobStorage()
+	fs := foremanServer{
+		serverContext: serverContext{jobStorage: &js, workingDirectory: determineWorkdir()},
+	}
+	pb.RegisterWorkerServer(s, &fs)
 	if err := s.Serve(l); err != nil {
 		log.Fatal(err)
 	}
 }
+
+func determineWorkdir() string {
+	workdir, workdirP := os.LookupEnv("FOREMAN_REX_WORKDIR")
```

Comment:
I'd consider `FOREMAN_YGG_WORKER_WORKDIR` or `FOREMAN_YGG_WORKDIR` as I think that would follow the convention that is being targeted for environment variable declaration of workers. e.g. https://github.com/theforeman/foreman-operations-collection/blob/develop/roles/cloud_connector/templates/foreman_rh_cloud.toml.j2#L3-L8

Although maybe going the route you have here with a change in the future to a naming scheme that is yggdrasil-worker-foreman-rex would better reflect what it is and then this would be in line with that already.

Reply:
Naming is hard, changed to the first option you proposed